### PR TITLE
Add Web UI authentication details

### PR DESCRIPTION
**Pull request by mosabua**

### CODE REVIEW

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 1]
```diff
@@ -16,7 +28,30 @@ running queries.
...
+
+The authentication type also supports usage of certificate authentication
+(``certificate``), :doc:`Kerberos authentication </security/server>`
+(``kerberos``), and JWT authentication (``jwt``).
+
+After successful login a user has access to all information of queries submitted
+to Presto under the same username.
+
+..
+    TBD .. detail how user can access all, also detail how more can be configured with file access control or whatever
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:49]: Not sure how both are done...

File: presto-docs/src/main/sphinx/admin/properties.rst [hunk 2]
```diff
@@ -862,11 +874,21 @@ The following properties can be used to configure the :doc:`./web-interface`.
...
     The duration how long a user can be logged into the Web UI, before the
     session times out, which forces an automatic log-out.
+
+``web-ui.user``
+^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Default value:**
+
+    The username automatically used for authentication to the UI with the
```
Comment (dain) [inline presto-docs/src/main/sphinx/admin/properties.rst:892]: maybe `s/to the UI/in the Web UI`.. should definitely be `Web UI to match the other references.

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 3]
```diff
@@ -8,6 +8,18 @@ HTTP/HTTPS, using the corresponding port number specified in the coordinator
 :ref:`config_properties`. It can be configured with :ref:`specific related
 properties <web-ui-properties>`.
 
+You can disable the Web UI, if you do not want to expose the information from
+the Web UI to your users.
```
Comment (dain) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:12]: maybe reference the property.

Also maybe something like:
```
The Web UI can be completely disabled if desired by setting the ``web-ui.enabled`` configuration property.
```

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 4]
```diff
@@ -16,7 +28,30 @@ running queries.
...
+type to ``fixed`` and configure the ``web-ui.user``. As a result any interaction
+in the Web UI, such as killing a running query, is logged with the configured
+user.
+
+The authentication type also supports usage of certificate authentication
+(``certificate``), :doc:`Kerberos authentication </security/server>`
+(``kerberos``), and JWT authentication (``jwt``).
+
+After successful login a user has access to all information of queries submitted
+to Presto under the same username.
```
Comment (dain) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:46]: I would state this as something like ".. user can view and kill their own queries. Query of other users may be visible if the user has permissions to view other users queries."

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 1 above)
Comment (dain) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:49]: https://prestosql.io/docs/current/security/built-in-system-access-control.html#query-rules

They would need "view" (in addition to "execute")

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 5]
```diff
@@ -44,7 +79,10 @@ For more detailed information about a query, simply click the query ID link.
 The query detail page has a summary section, graphical representation of various stages of the
 query and a list of tasks. Each task ID can be clicked to get more information about that task.
 
-The summary section has a button to kill the currently running query. There are two visualizations
+The summary section has a button to kill the currently running query. Users can
+only kill queries, submitted with their username.
```
Comment (dain) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:83]: Same as with view.  They can have permissions to kill other users queries.

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 6]
```diff
@@ -16,7 +28,30 @@ running queries.
 When the server is configured to use HTTPS, a :doc:`password authenticator
 </develop/password-authenticator>` such as :doc:`LDAP </security/ldap>` or
 :doc:`password file </security/password-file>` must be configured in order to
-use the Web UI.
+use the Web UI. In these cases, the authentication type is automatically
+defaulted to the required ``form`` value. It triggers the Web UI to display a
+login form that accepts the relevant username and password and verifies against
+the correct authentication backend.
```
Comment (dain) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:34]: I find these sections confusing.  There are now multiple authentication systems supported, so I feel `must` is too strong here.  Also in the following sections is says `also supports usage of certificate authentication`, which to me sounds like I can use form and certificate, but these are alternatives.

File: presto-docs/src/main/sphinx/admin/properties.rst (hunk 2 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/properties.rst:892]: Fixed

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 3 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:12]: Added to my sentence.

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 4 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:46]: Great... I wasn't sure how it works..

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 1 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:49]: Updated with link to this.

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 6 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:33]: Updated.

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 7]
```diff
@@ -16,7 +27,30 @@ running queries.
...
+use the Web UI. In these cases, the authentication type defaults to the required
+``form`` value. It triggers the Web UI to display a login form, that accepts the
+relevant username and password, and verifies against the authentication backend.
+
+If you require the Web UI to be accessible to anyone, you can set the
+authentication type to ``fixed`` and configure the ``web-ui.user``. As a result
+any interaction in the Web UI, such as killing a running query, is logged with
+the configured user.
+
+Alternatively the authentication type can be set to u se certificate
```
Comment (dain) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:39]: extra space

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 8]
```diff
@@ -16,7 +27,30 @@ running queries.
...
+authentication (``certificate``), :doc:`Kerberos authentication
+</security/server>` (``kerberos``), or JWT authentication (``jwt``).
+
+After successful login you have access to all information of queries submitted
+to Presto under the same username. You can view and kill your own queries.
+Queries of other users may be visible, if you have permissions to view other
+users queries. You can use :ref:`query rules <query_rules>` with the built-in
+access control to configure access to viewing, executing, as well as killing
```
Comment (dain) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:47]: I'd drop execute here as the UI can't execute queries

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 9]
```diff
@@ -16,7 +27,30 @@ running queries.
...
+authentication type to ``fixed`` and configure the ``web-ui.user``. As a result
+any interaction in the Web UI, such as killing a running query, is logged with
+the configured user.
+
+Alternatively the authentication type can be set to u se certificate
+authentication (``certificate``), :doc:`Kerberos authentication
+</security/server>` (``kerberos``), or JWT authentication (``jwt``).
+
+After successful login you have access to all information of queries submitted
+to Presto under the same username. You can view and kill your own queries.
```
Comment (dain) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:44]: maybe combine these two sentence...  I'm thinking mixing the information in to the first part and not using a conjunction.

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 7 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:39]: weird how that popped in. fixed

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 8 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:47]: done

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 9 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:44]: done and simplified rest of paragraph as well

File: presto-docs/src/main/sphinx/admin/properties.rst [hunk 10]
```diff
@@ -843,6 +843,18 @@ Web UI Properties
 
 The following properties can be used to configure the :doc:`./web-interface`.
 
+``web-ui.authentication.type``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Allowed values:** ``form``, ``fixed``, ``certificate``, ``kerberos``,
+      ``jwt``
```
Comment (electrum) [inline presto-docs/src/main/sphinx/admin/properties.rst:851]: Don't wrap a single value, as this is harder to read

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 11]
```diff
@@ -8,6 +8,17 @@ HTTP/HTTPS, using the corresponding port number specified in the coordinator
 :ref:`config_properties`. It can be configured with :ref:`specific related
 properties <web-ui-properties>`.
 
+You can disable the Web UI with the ``web-ui.enabled`` property, if you do not
+want to expose the information from the Web UI to your users.
+
+.. contents::
```
Comment (electrum) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:14]: No need for this, since the page is short.

File: presto-docs/src/main/sphinx/admin/properties.rst [hunk 12]
```diff
@@ -843,6 +843,18 @@ Web UI Properties
...
+    * **Default value:** ``form``
+
+    The authentication mechanism to allow user access to the Web UI. Find more
+    informtion in the :ref:`Web UI authenication section
```
Comment (electrum) [inline presto-docs/src/main/sphinx/admin/properties.rst:855]: Typos "informtion" and "authenication". These typos are duplicated below.

Shorten this to
```
See :ref:`Web UI Authentication <web-ui-authentication>`.

File: presto-docs/src/main/sphinx/admin/web-interface.rst [hunk 13]
```diff
@@ -8,6 +8,17 @@ HTTP/HTTPS, using the corresponding port number specified in the coordinator
 :ref:`config_properties`. It can be configured with :ref:`specific related
 properties <web-ui-properties>`.
 
+You can disable the Web UI with the ``web-ui.enabled`` property, if you do not
```
Comment (electrum) [inline presto-docs/src/main/sphinx/admin/web-interface.rst:11]: No need for comma

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 11 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst:14]: this is mostly in anticipation, but also because the page is longer than one screen at this stage. The page needs a LOT more info, but I removed for now.

File: presto-docs/src/main/sphinx/admin/properties.rst (hunk 12 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/properties.rst:855]: Done in both places.

File: presto-docs/src/main/sphinx/admin/web-interface.rst (hunk 5 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/admin/web-interface.rst]: I removed that since its covered earlier.

**Comment by electrum** [comment]
Merged, thanks!
